### PR TITLE
Remove explicit GC calls in the asset cache

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This is a holdover from our 32bit days where we needed to be much more aggressive about relinquishing memory.  We do not need that now, and we want to let the normal GC do its thing.

### CODE REVIEW

File: src/VisualStudio/Core/Test.Next/Services/SolutionAssetCacheTests.cs [hunk 1]
```diff
@@ -20,6 +20,12 @@ namespace Roslyn.VisualStudio.Next.UnitTests.Remote
     [Trait(Traits.Feature, Traits.Features.RemoteHost)]
     public class SolutionAssetCacheTests
     {
+        private static void ForceGC()
+        {
+            for (var i = 0; i < 3; i++)
+                GC.Collect();
+        }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Test.Next/Services/SolutionAssetCacheTests.cs:27]: this is a move of waht the product code used to do.  i moved it here so test code dosen't have to update.

File: src/Workspaces/Remote/ServiceHub/Host/SolutionAssetCache.cs [hunk 2]
```diff
@@ -138,42 +115,10 @@ private async Task CleanAssetsAsync()
         while (!cancellationToken.IsCancellationRequested)
         {
             await CleanAssetsWorkerAsync(cancellationToken).ConfigureAwait(false);
-
-            ForceGC();
```
Comment (ToddGrun) [inline src/Workspaces/Remote/ServiceHub/Host/SolutionAssetCache.cs:142]: >ForceGC(); [](http://example.com/codeflow?start=12&length=10)

I can do this in a separate PR, but it kinda bugs me that this has it's own dedicated thread for this. ABWQ usage would prevent the need for that, right?

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/Remote/ServiceHub/Host/SolutionAssetCache.cs (hunk 2 above)
Comment (ToddGrun) [reply src/Workspaces/Remote/ServiceHub/Host/SolutionAssetCache.cs:142]: Never mind, expanded my search and there are other activities on this thread

File: src/VisualStudio/Core/Test.Next/Services/SolutionAssetCacheTests.cs [hunk 3]
```diff
@@ -20,6 +20,12 @@ namespace Roslyn.VisualStudio.Next.UnitTests.Remote
     [Trait(Traits.Feature, Traits.Features.RemoteHost)]
     public class SolutionAssetCacheTests
     {
+        private static void ForceGC()
+        {
+            for (var i = 0; i < 3; i++)
+                GC.Collect();
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Test.Next/Services/SolutionAssetCacheTests.cs:26]: If this is being used for testing it's odd it isn't doing the WaitForPendingFinalizers that is often used....

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Test.Next/Services/SolutionAssetCacheTests.cs:26]: Yes.  thsi is very odd.

**Pull request merged**